### PR TITLE
APIv4 - Support updating activity.case_id

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Allows updating the case(s) associated with an activity. This allows e.g. a SearchKit display of activities to expose "File On Case" as an in-place-editable column.

Before
----------------------------------------
The API supported filing a brand new activity on a case (during create action).

After
----------------------------------------
Unchanged for APIv3.
APIv4 now supports changing the case associated with an activity; switching it to a different case or removing it altogether. 

Technical Details
----------------------------------------
Care was taken to support the edge case of multiple cases per activity while maintaining backward-compatibility.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
[Test Result](https://test.civicrm.org/job/CiviCRM-Core-PR/50935/testReport/) (3 failures / +2)
[CRM_Case_BAO_CaseTest.testGetRelatedCases](https://test.civicrm.org/job/CiviCRM-Core-PR/50935/testReport/junit/(root)/CRM_Case_BAO_CaseTest/testGetRelatedCases/)
[CRM_Case_BAO_CaseTest.testChangeCaseStatusLinkedCases](https://test.civicrm.org/job/CiviCRM-Core-PR/50935/testReport/junit/(root)/CRM_Case_BAO_CaseTest/testChangeCaseStatusLinkedCases/)
[api\v4\Action\NullValueTest.testSaveWithReload](https://test.civicrm.org/job/CiviCRM-Core-PR/50935/testReport/junit/(root)/api_v4_Action_NullValueTest/testSaveWithReload/)

File: CRM/Case/Form/Activity/LinkCases.php [hunk 1]
```diff
@@ -145,7 +145,7 @@ public static function endPostProcess($form, $params, $activity) {
     //create a link between two cases.
     if ($activityId && $linkCaseID) {
       $caseParams = [
-        'case_id' => $linkCaseID,
+        'case_id' => array_merge($form->_caseId, [$linkCaseID]),
```
Comment (demeritcowboy) [inline CRM/Case/Form/Activity/LinkCases.php:148]: This doesn't work because processCaseActivity is expecting an int. It ends up linking every case to case id 1. And because the test down below has been changed to just test the activity api instead of simulating the form it doesn't catch this (admittedly it wasn't a great simulation, but called the form function).

Comment (demeritcowboy) [reply CRM/Case/Form/Activity/LinkCases.php:148]: Better unit test of the form: https://github.com/civicrm/civicrm-core/pull/24328

Comment (colemanw) [reply CRM/Case/Form/Activity/LinkCases.php:148]: Thanks @demeritcowboy but I don't think `processCaseActivity` actually gets called from here. This PR changes the Activity BAO to call a new function that can handle an array.

Comment (demeritcowboy) [reply CRM/Case/Form/Activity/LinkCases.php:148]: But just run it and you'll see it does, and links it to case id 1.

Comment (colemanw) [reply CRM/Case/Form/Activity/LinkCases.php:148]: Ok thanks I'll have a look
